food_delivery/foodapp/views.py

```diff
@@ -1,3 +1,453 @@
-from django.shortcuts import render
+from .models import *
+from .serializers import * 
+from rest_framework import status
+from rest_framework.views import APIView
+from geopy.geocoders import Nominatim
+from geopy.distance import geodesic
+from datetime import datetime, timedelta
+from rest_framework.response import Response
 
-# Create your views here.
+class UserListCreate(APIView):
+    def get(self, request):
+        users = User.objects.all()
+        serializer = UserSerializer(users, many=True)
+        return Response(serializer.data)
+
+    def post(self, request):
+        serializer = UserSerializer(data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+class UserDetail(APIView):
+    def get_object(self,pk):
+        try:
+            return User.objects.get(pk=pk)
+        except User.DoesNotExist:
+            return None
+        
+    def get(self,request,pk):
+        user=self.get_object(pk)
+        if user is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        serializer = UserSerializer(user)
+        return Response(serializer.data)
+
+    def put(self, request, pk):
+        user = self.get_object(pk)
+        if user is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        serializer = UserSerializer(user, data=request.data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+    def delete(self, request, pk):
+        user = self.get_object(pk)
+        if user is None:
+            return Response(status=status.HTTP_404_NOT_FOUND)
+        user.delete()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+class RestaurantListCreate(APIView):
+    def get(self, request):
+        restaurants = Restaurant.objects.all()
```

Comment:
you are not checking whether restaurant is active, is available. you are using .all() in many places in get API. try to avoid it and start using some filters. .all() should be used only in rare cases.